src/field_theory/fixed.lean

```diff
@@ -159,3 +163,110 @@ by { rw [← fixed_points.minpoly.minimal_polynomial,
+    ⟨i, his⟩ (mem_attach _ _) : _),
+  refine (@sum_attach _ _ s _ (λ i, (g • l i - l i) • (to_fun G F) i)).trans _, ext g', dsimp only,
+  conv_lhs { rw sum_apply, congr, skip, funext, rw [pi.smul_apply, sub_smul, smul_eq_mul] },
+  rw [sum_sub_distrib, pi.zero_apply, sub_eq_zero],
+  conv_lhs { congr, skip, funext,
+    rw [to_fun_apply, ← mul_inv_cancel_left g g', mul_smul, ← smul_mul', ← to_fun_apply _ x] },
+  show ∑ x in s, g • (λ y, l y • to_fun G F y) x (g⁻¹ * g') =
+    ∑ x in s, (λ y, l y • to_fun G F y) x g',
+  rw [← smul_sum, ← sum_apply _ _ (λ y, l y • to_fun G F y),
+      ← sum_apply _ _ (λ y, l y • to_fun G F y)], dsimp only,
+  rw [hla, to_fun_apply, to_fun_apply, smul_smul, mul_inv_cancel_left]
+end
+
+variables [fintype G]
+
+lemma fixed_points.dim_le_card : vector_space.dim (fixed_points G F) F ≤ fintype.card G :=
+begin
+  refine dim_le (λ s hs, cardinal.nat_cast_le.1 _),
+  rw [← @dim_fun' F G, ← cardinal.lift_nat_cast.{v (max u v)},
+    cardinal.finset_card, ← cardinal.lift_id (vector_space.dim F (G → F))],
+  exact linear_independent_le_dim'.{_ _ _ (max u v)}
+    (linear_independent_smul_of_linear_independent G F hs)
+end
+
+instance finite_dimensional_fixed_points : finite_dimensional (fixed_points G F) F :=
+finite_dimensional.finite_dimensional_iff_dim_lt_omega.2 $
+lt_of_le_of_lt (fixed_points.dim_le_card G F) (cardinal.nat_lt_omega _)
+
+lemma findim_fixed_points : findim (fixed_points G F) F ≤ fintype.card G :=
+by exact_mod_cast trans_rel_right (≤) (findim_eq_dim _ _) (fixed_points.dim_le_card G F)
+
+lemma linear_independent_to_linear_map (R : Type u) (A : Type v)
+  [comm_semiring R] [integral_domain A] [algebra R A] :
+  linear_independent A (alg_hom.to_linear_map : (A →ₐ[R] A) → (A →ₗ[R] A)) :=
+have linear_independent A (linear_map.lto_fun R A A ∘ alg_hom.to_linear_map),
+from ((linear_independent_monoid_hom A A).comp
+  (coe : (A →ₐ[R] A) → (A →* A))
+  (λ f g hfg, alg_hom.ext $ monoid_hom.ext_iff.1 hfg) : _),
+linear_independent_of_comp _ this
+
+lemma cardinal_mk_alg_hom (K : Type u) (V : Type v)
+  [field K] [field V] [algebra K V] [finite_dimensional K V] :
+  cardinal.mk (V →ₐ[K] V) ≤ findim V (V →ₗ[K] V) :=
+cardinal_mk_le_findim_of_linear_independent $ linear_independent_to_linear_map K V
+
+noncomputable instance alg_hom.fintype (K : Type u) (V : Type v)
+  [field K] [field V] [algebra K V] [finite_dimensional K V] :
+  fintype (V →ₐ[K] V) :=
+classical.choice $ cardinal.lt_omega_iff_fintype.1 $
+lt_of_le_of_lt (cardinal_mk_alg_hom K V) (cardinal.nat_lt_omega _)
+
+lemma findim_alg_hom (K : Type u) (V : Type v)
+  [field K] [field V] [algebra K V] [finite_dimensional K V] :
+  fintype.card (V →ₐ[K] V) ≤ findim V (V →ₗ[K] V) :=
+fintype_card_le_findim_of_linear_independent $ linear_independent_to_linear_map K V
+
+/-- Embedding produced from a faithful action. -/
+def to_alg_hom_over_fixed_points (G : Type u) (F : Type v) [group G] [field F]
```

Comment:
```suggestion
@[simps]
def to_alg_hom_over_fixed_points (G : Type u) (F : Type v) [group G] [field F]
```

Reply:
It doesn't generate the correct lemmas (see below).